lib/_http_outgoing.js

```diff
@@ -354,6 +372,17 @@ function _writeRaw(data, encoding, callback) {
     encoding = null;
   }
 
+  let byteLen;
+  if (typeof data === 'string') {
+    byteLen = Buffer.byteLength(data, encoding);
+  } else {
+    byteLen = data.length;
+  }
+  if (byteLen + this[kBytesWritten] > this[kSentContentLength]) {
+    const contentLen = this[kSentContentLength];
+    throw new ERR_HTTP_INVALID_HEADER_VALUE('Content-Length', contentLen);
```

Comment:
I don't think the header value is wrong per per se. It's more of a mistmatch.

Reply:
header names in node supposed to be lower case always as a convention.